syntax="proto3";

import "google/protobuf/empty.proto";
import "protos/data.proto";
import "protos/controller.proto";
import "protos/data_bundle.proto";
import "protos/exchange.proto";

service Hub {
    rpc UploadGraph(stream Data) returns (google.protobuf.Empty);
    rpc DownloadGraph(google.protobuf.Empty) returns (stream Data);
    rpc GetController(ControllerRequest) returns (ControllerReply);
    rpc GetDataBundle(CompoundDomainDef) returns (stream Data);
    rpc Deploy(DeploymentRequest) returns (DeploymentResponse);
    rpc GetCalendars(google.protobuf.Empty) returns (stream Data);
    rpc GetExchanges(google.protobuf.Empty) returns (stream Exchange);
}

message DeploymentRequest {

}

message DeploymentResponse {

}